### PR TITLE
Upgrade MW & Wikibase configuration for MW 1.37

### PR DESCRIPTION
**Pull request by deer-wmde**
This PR updates our configuration of MW & Wikibase according to the 1.37 Release Notes of them.

I couldn't find any complications with the MW config itself, but there are some changes needed for Wikibase.

### CODE REVIEW

**Comment by deer-wmde** [comment]
There is this script mentioned in the Wikibase release notes. I ran it and it gave me this suggested configuration:

```php
// Suggested entitySource configuration to be added in LocalSettings.php:

$wgWBRepoSettings['entitySources'] = array (
  'local' => 
  array (
    'entityNamespaces' => 
    array (
      'item' => '120/main',
      'property' => '122/main',
      'lexeme' => '146/main',
    ),
    'repoDatabase' => false,
    'baseUri' => 'http://maintenance/entity/',
    'rdfNodeNamespacePrefix' => 'wd',
    'rdfPredicateNamespacePrefix' => '',
    'interwikiPrefix' => '',
  ),
);

$wgWBClientSettings['entitySources'] = array (
  'local' => 
  array (
    'entityNamespaces' => 
    array (
      'item' => '120/main',
      'property' => '122/main',
    ),
    'repoDatabase' => false,
    'baseUri' => 'http://maintenance/entity/',
    'rdfNodeNamespacePrefix' => 'wd',
    'rdfPredicateNamespacePrefix' => '',
    'interwikiPrefix' => '',
  ),
);

```

**Comment by addshore** [comment]
Looks like the baseUri in that snippet would need dynamically setting.
And also 'rdfNodeNamespacePrefix' might be worth us at least scratching our heads over to make sure we are doing the right thing there.

File: wbstack/src/Settings/LocalSettings.php [hunk 1]
```diff
@@ -235,11 +235,6 @@
     ];
 }
 
-// Output compression needs to be disabled in 1.35 until the below phab task is fixed...
-// TODO dig more to see if there is something else to do here...
-// https://phabricator.wikimedia.org/T235554
-$wgDisableOutputCompression  = true;
```
Comment (deer-wmde) [inline wbstack/src/Settings/LocalSettings.php:241]: This issue seems to be fixed, I tested it locally with something like http://potatopotato.wbaas.localhost/wiki/Special:EntityData/Q1.json and it returned raw JSON

File: wbstack/src/Settings/LocalSettings.php [hunk 2]
```diff
@@ -493,7 +489,7 @@ function onPersonalUrlsConfirmAccount( array &$personal_urls, Title $title, Skin
 require_once "$IP/extensions/Wikibase/client/ExampleSettings.php";
 
 // Force the concept URIs to be http (as this has always been the way on wbstack)
-$wgWBRepoSettings['conceptBaseUri'] = 'http://' . $wikiInfo->domain . '/entity/';
+$wgWBBaseUri = 'http://' . $wikiInfo->domain . '/entity/';
```
Comment (deer-wmde) [inline wbstack/src/Settings/LocalSettings.php:492]: [* (repo) (T287486) conceptBaseUri has been removed.](https://github.com/wikimedia/Wikibase/blob/master/RELEASE-NOTES-1.37)

File: wbstack/src/Settings/LocalSettings.php [hunk 3]
```diff
@@ -515,19 +511,39 @@ function onPersonalUrlsConfirmAccount( array &$personal_urls, Title $title, Skin
...
-        'baseUri' => $wgWBRepoSettings['conceptBaseUri'],
-        'entityNamespaces' => [
-            'item' => 120,
-            'property' => 122,
-        ],
-        'prefixMapping' => [ '' => '' ],
-    ]
-];
+
+$wgWBRepoSettings['entitySources'] = array (
```
Comment (deer-wmde) [inline wbstack/src/Settings/LocalSettings.php:515]: [* (repo+client) (T257262) The entitySources setting is no longer optional, but required.](https://github.com/wikimedia/Wikibase/blob/master/RELEASE-NOTES-1.37)
[* (client) (T285471) repositories has been removed.](https://github.com/wikimedia/Wikibase/blob/master/RELEASE-NOTES-1.37)

File: wbstack/src/Settings/LocalSettings.php (hunk 2 above)
Comment (jakobw) [reply wbstack/src/Settings/LocalSettings.php:492]: The `wg` in the name is for globals, and this isn't a global. I'm pretty sure we can remove the comment above and just move this above the entitySources definition and maybe call the variable something like `$localConceptBaseUri`.

**Review by jakobw — CHANGES_REQUESTED** [review]
Found some details to nitpick about, otherwise this is looking pretty good!

**Comment by deer-wmde** [comment]
> Found some details to nitpick about, otherwise this is looking pretty good!

Nice thanks, good calls! I think I fixed everything you mentioned now.

**Review by jakobw — APPROVED** [review]
Looks good to me :tada:

**Pull request merged**